ts/db_test.go

```diff
@@ -0,0 +1,351 @@
+// data matches.
+type tsTestModel struct {
+	t         testing.TB
+	modelData map[string]*proto.Value
+	db        *DB
+
+	client      *client.KV
+	engine      engine.Engine
+	store       *storage.Store
+	transport   multiraft.Transport
+	manualClock *hlc.ManualClock
+	clock       *hlc.Clock
+	stopper     *util.Stopper
+}
+
+// newTsTestModel creates a new tsTestModel instance. The Start() method must
+// be called before using it.
+func newTsTestModel(t *testing.T) *tsTestModel {
+	return &tsTestModel{
+		t:         t,
+		modelData: make(map[string]*proto.Value),
+	}
+}
+
+// Start constructs and starts the ts.DB instance and the cockroach store
+// backing it.
+func (tm *tsTestModel) Start() {
+	// Initialize some supporting objects needed to initialize the store.
+	rpcContext := rpc.NewContext(hlc.NewClock(hlc.UnixNano), rpc.LoadInsecureTLSConfig())
+	gossip := gossip.New(rpcContext, gossip.TestInterval, gossip.TestBootstrap)
+	tm.manualClock = hlc.NewManualClock(0)
+	tm.clock = hlc.NewClock(tm.manualClock.UnixNano)
+	tm.transport = multiraft.NewLocalRPCTransport()
+	tm.stopper = util.NewStopper()
+	tm.stopper.AddCloser(tm.transport)
+
+	// Initialize a new in-memory engine.
+	tm.engine = engine.NewInMem(proto.Attributes{Attrs: []string{"dc1", "mem"}}, 1<<20)
+
+	// Initialize and bootstrap a store which uses the engine.
+	sender := kv.NewLocalSender()
+	tm.store = storage.NewStore(tm.clock, tm.engine, client.NewKV(nil, sender), gossip, tm.transport, storage.TestStoreConfig)
+	if err := tm.store.Bootstrap(proto.StoreIdent{NodeID: 1, StoreID: 1}, tm.stopper); err != nil {
+		tm.t.Fatal(err)
+	}
+	if err := tm.store.BootstrapRange(); err != nil {
+		tm.t.Fatal(err)
+	}
+	sender.AddStore(tm.store)
+	if err := tm.store.Start(tm.stopper); err != nil {
+		tm.t.Fatal(err)
+	}
+
+	tm.initConfigs()
+
+	// Initialize a client which talks to the store.
+	tm.client = client.NewKV(nil, sender)
+
+	// Initialize the DB instance.
+	tm.db = NewDB(client.NewKV(nil, sender))
+}
+
+// Stop stops the system under test.
+func (tm *tsTestModel) Stop() {
+	tm.stopper.Stop()
+}
+
+// initConfigs adds some default configuration entries to the data store.
+func (tm *tsTestModel) initConfigs() {
+	if err := engine.MVCCPutProto(tm.engine, nil, engine.KeyConfigAccountingPrefix, proto.MinTimestamp, nil, &testDefaultAcctConfig); err != nil {
+		tm.t.Fatal(err)
+	}
+	if err := engine.MVCCPutProto(tm.engine, nil, engine.KeyConfigPermissionPrefix, proto.MinTimestamp, nil, &testDefaultPermConfig); err != nil {
+		tm.t.Fatal(err)
+	}
+	if err := engine.MVCCPutProto(tm.engine, nil, engine.KeyConfigZonePrefix, proto.MinTimestamp, nil, &testDefaultZoneConfig); err != nil {
+		tm.t.Fatal(err)
+	}
+}
+
+// getActualData returns the actual value of all time series keys in the
+// underlying engine. Data is returned as a map of strings to proto.Values.
+func (tm *tsTestModel) GetActualData() map[string]*proto.Value {
```

Comment:
I thought we agreed that we could capitalize methods of unexported structures, for the sake of clarity.